### PR TITLE
Adding option to track resources upwards, rather than downward

### PR DESCRIPTION
**Pull request by noxora**
This provides a configuration value in the resource tracking section which allows the counter to move upward rather than down

While the tracker is set to upward: 
All counters with a configuration value greater than 0 are initialized at 0 at the start of the gauntlet and count up as the player collects them. The text of the counter turns green when the player has collected their configured amount of resources

While the tracker is set to downward:
Current behavior is unchanged

### CODE REVIEW

File: src/main/java/ca/gauntlet/module/maze/ResourceCounter.java [hunk 1]
```diff
@@ -46,6 +51,10 @@ class ResourceCounter extends InfoBox
 	private int count;
 	private String text;
 
+	@Inject
```
Comment (noxora) [inline src/main/java/ca/gauntlet/module/maze/ResourceCounter.java:54]: This is currently broken and doesn't get populated, resulting in a null pointer in `resourceTarget()` below

Comment (noxora) [reply src/main/java/ca/gauntlet/module/maze/ResourceCounter.java]: This has been fixed

**Comment by noxora** [comment]
If someone could help me find a good way to only have the tracking boxes appear as resources are collected, rather than all at once at the beginning, that would be ideal. Otherwise this is good to go

File: src/main/java/ca/gauntlet/module/maze/ResourceCounter.java [hunk 2]
```diff
@@ -113,4 +125,30 @@ private static InfoBoxPriority getPriority(final Resource resource)
...
+				return config.resourceTirinum();
+			case GRYM_LEAF:
+			case CORRUPTED_GRYM_LEAF:
+				return config.resourceGrym();
+			case CRYSTAL_SHARDS:
+			case CORRUPTED_SHARDS:
+				return config.resourceShard();
+			case RAW_PADDLEFISH:
+				return config.resourcePaddlefish();
+			case WEAPON_FRAME:
```
Comment (rdutta) [inline src/main/java/ca/gauntlet/module/maze/ResourceCounter.java:148]: Missing case for corrupted weapon frame (?)

**Review by rdutta — CHANGES_REQUESTED** [review]
Can you elaborate on your issue comment "fill the gap in functionality"? What is the benefit of incrementing the counter instead of decrementing it?

Related, there is a config `Dynamically remove overlays` that needs to be updated to handle incremental counting if this feature is accepted.

File: src/main/java/ca/gauntlet/module/maze/ResourceCounter.java (hunk 2 above)
Comment (noxora) [reply src/main/java/ca/gauntlet/module/maze/ResourceCounter.java:148]: That's correct, I need to fix that

**Comment by noxora** [comment]
The "gap in functionality" is created by the plugin "Gauntlet" (https://runelite.net/plugin-hub/show/gauntlet) which many players use to track resources collected, rather than count down resource needs. Multiple guides that I watched online for gauntlet recommended installing both this plugin and the "Gauntlet" plugin, as well as Hunllef Helper, so I would assume they are often all downloaded together, and without any configuration of any of the plugins, the defaults appear to work well, with the highlighting and timer provided by this plugin, "Gauntlet" tracking resources collected in the maze and Hunllef helper keeping track in the fight. 

The sticking point is that "Gauntlet" hasn't been updated since May, 2020, and has multiple open issues, including the bug I initially reported here, thinking that this plugin was the cause. Since I believe "Gauntlet" to be unmaintained at this point, I think it would be good to offer players a working alternative to the functionality it provided.

**Comment by noxora** [comment]
I've committed changes to add the dynamic highlighting functionality

**Review by rdutta — APPROVED** [review]

**Comment by rdutta** [comment]
> The gap in functionality is created by the plugin "Gauntlet" (https://runelite.net/plugin-hub/show/gauntlet) which many players use to track resources collected, rather than count down resource needs. Multiple guides that I watched online for gauntlet recommended installing both this plugin and the "Gauntlet" plugin, as well as Hunllef Helper, so I would assume they are often all downloaded together, and without any configuration of any of the plugins, the defaults appear to work well, with the highlighting and timer provided by this plugin, "Gauntlet" tracking resources collected in the maze and Hunllef helper keeping track in the fight.
> 
> The sticking point is that "Gauntlet" hasn't been updated since May, 2020, and has multiple open issues, including the bug I initially reported here, thinking that this plugin was the cause. Since I believe "Gauntlet" to be unmaintained at this point, I think it would be good to offer players a working alternative to the functionality it provided.

I see that repo has not been updated in 3 years, so it might be best to simply replicate the feature for now. I have pushed some changes that keep the infobox counter color white and only creates boxes when a resource is collected. Any resource with a value of '0' will not be tracked in either tracking mode.

If you agree with these changes and it works as expected (please test) I can accept the PR.

**Comment by noxora** [comment]
I agree with these changes, I noticed that dynamically removing overlays is now set to work only for decrement mode. As long as this is okay with you I'm happy. I tested the plugin in both regular and corrupted gauntlet, on both increment and decrement modes for each, and it seems to work as intended, with the behavior that we've talked about. Thank you for your help

**Comment by while-loop** [comment]
Just digging through this plugin's repo. 
I'm [disabling](https://github.com/runelite/plugin-hub/pull/4963) the old Gaunlet plugin as this one is much better and includes much more features for players. Thanks for adding feature parity :+1:

**Pull request merged**